Class alias

builtin/maybe.sk should have these
```
Some = Maybe::Some
None = Maybe::None
```
but currently results in this error.
```
cargo run -- run examples/a.sk
    Finished dev [unoptimized + debuginfo] target(s) in 0.09s
     Running `target/debug/shiika run examples/a.sk`
Undefined symbols for architecture x86_64:
  "_init_::None", referenced from:
      _init_constants in a-cad86a.o
  "_init_::Some", referenced from:
      _init_constants in a-cad86a.o
```

--- a/builtin/_lambda.sk
+++ b/builtin/_lambda.sk
@@ -40,6 +40,3 @@ end
 
 class Fn9<S1, S2, S3, S4, S5, S6, S7, S8, S9, T> : Fn
 end
-
-# Create `::FnX` by these ConstRef's for lambda exprs (#178)
-Fn0; Fn1; Fn2; Fn3; Fn4; Fn5; Fn6; Fn7; Fn8; Fn9

--- a/builtin/array.sk
+++ b/builtin/array.sk
@@ -385,6 +385,3 @@ class Array<T>
     l
   end
 end
-
-# Create `Meta:Array` by this ConstRef for array literals (#178)
-Array

--- a/builtin/maybe.sk
+++ b/builtin/maybe.sk
@@ -2,5 +2,5 @@ enum Maybe<V>
   case Some(value: V)
   case None
 end
-#Some = Maybe::Some
-#None = Maybe::None
+Some = Maybe::Some
+None = Maybe::None

--- a/builtin/result.sk
+++ b/builtin/result.sk
@@ -1,4 +1,6 @@
 enum Result<V, E>
   case Ok(value: V)
-  case Err(err: E)
+  case Fail(err: E)
 end
+Ok = Result::Ok
+Fail = Result::Fail

--- a/src/ast.rs
+++ b/src/ast.rs
@@ -175,6 +175,18 @@ impl AstExpression {
             _ => false,
         }
     }
+
+    /// If `self` is ConstAssign, convert it to a ConstDefinition
+    pub fn as_const_def(&self) -> Option<Definition> {
+        if let AstExpressionBody::ConstAssign { names, rhs } = &self.body {
+            Some(Definition::ConstDefinition {
+                name: names.join("::"),
+                expr: *rhs.clone(),
+            })
+        } else {
+            None
+        }
+    }
 }
 
 pub fn logical_not(expr: AstExpression) -> AstExpression {

--- a/src/code_gen/mod.rs
+++ b/src/code_gen/mod.rs
@@ -113,11 +113,12 @@ impl<'hir: 'ictx, 'run, 'ictx: 'run> CodeGen<'hir, 'run, 'ictx> {
         self.gen_methods(&hir.sk_methods)?;
         self.gen_const_inits(&hir.const_inits)?;
         if self.generate_main {
-            self.gen_init_constants(&hir.const_inits, &imports.constants);
+            self.gen_init_constants(&hir.const_inits, true);
             self.gen_user_main(&hir.main_exprs, &hir.main_lvars)?;
             self.gen_main();
         } else {
             // generating builtin
+            self.gen_init_constants(&hir.const_inits, false);
             self.impl_boxing_funcs();
         }
         self.gen_lambda_funcs(hir)?;
@@ -261,24 +262,29 @@ impl<'hir: 'ictx, 'run, 'ictx: 'run> CodeGen<'hir, 'run, 'ictx> {
 
     /// Generate `init_constants()`
     // TODO: imported_constants should be Vec (order matters)
-    fn gen_init_constants(
-        &self,
-        const_inits: &'hir [HirExpression],
-        imported_constants: &HashMap<ConstFullname, TermTy>,
-    ) {
-        // define void @init_constants()
+    fn gen_init_constants(&self, const_inits: &'hir [HirExpression], is_main: bool) {
+        let package_name = if is_main { "main" } else { "builtin" };
+        // define void @xxx_init_constants()
         let fn_type = self.void_type.fn_type(&[], false);
-        let function = self.module.add_function("init_constants", fn_type, None);
+        let function =
+            self.module
+                .add_function(&format!("{}_init_constants", package_name), fn_type, None);
         let basic_block = self.context.append_basic_block(function, "");
         self.builder.position_at_end(basic_block);
 
-        // call void @"init_::XX"()
-        for fullname in imported_constants.keys() {
-            if !fullname.is_internal() {
-                let func = self.get_llvm_func(&format!("init_{}", fullname.0));
+        // Initialize imported constants
+        if is_main {
+            let imports = vec!["builtin"];
+            for s in imports {
+                let fn_type = self.void_type.fn_type(&[], false);
+                self.module
+                    .add_function(&format!("{}_init_constants", s), fn_type, None);
+                let func = self.get_llvm_func(&format!("{}_init_constants", s));
                 self.builder.build_call(func, &[], "");
             }
         }
+
+        // Initialize own constants
         for expr in const_inits {
             match &expr.node {
                 HirExpressionBase::HirConstAssign { fullname, .. } => {
@@ -336,7 +342,7 @@ impl<'hir: 'ictx, 'run, 'ictx: 'run> CodeGen<'hir, 'run, 'ictx> {
         self.builder.build_call(func, &[], "");
 
         // Call init_constants, user_main
-        let func = self.get_llvm_func("init_constants");
+        let func = self.get_llvm_func("main_init_constants");
         self.builder.build_call(func, &[], "");
         let func = self.get_llvm_func("user_main");
         self.builder.build_call(func, &[], "");

--- a/src/parser/definition_parser.rs
+++ b/src/parser/definition_parser.rs
@@ -539,7 +539,7 @@ impl<'a> Parser<'a> {
         Ok(ConstName { names, args })
     }
 
-    fn parse_const_definition(&mut self) -> Result<ast::Definition, Error> {
+    pub fn parse_const_definition(&mut self) -> Result<ast::Definition, Error> {
         self.debug_log("parse_const_definition");
         self.lv += 1;
         let name;

--- a/src/parser/mod.rs
+++ b/src/parser/mod.rs
@@ -77,12 +77,19 @@ impl<'a> Parser<'a> {
                     items.push(ast::TopLevelItem::Def(self.parse_enum_definition()?));
                 }
                 Token::KwDef => {
-                    // REFACTOR: Raise error here (rather than on ClassDict::index_program)
-                    items.push(ast::TopLevelItem::Def(self.parse_method_definition()?));
+                    return Err(parse_error!(
+                        self,
+                        "you cannot define toplevel method in Shiika"
+                    ));
                 }
                 Token::Eof | Token::KwEnd => break,
                 _ => {
-                    items.push(ast::TopLevelItem::Expr(self.parse_expr()?));
+                    let expr = self.parse_expr()?;
+                    if let Some(constdef) = expr.as_const_def() {
+                        items.push(ast::TopLevelItem::Def(constdef));
+                    } else {
+                        items.push(ast::TopLevelItem::Expr(expr));
+                    }
                 }
             }
             self.skip_wsn();

--- a/tests/expression_parser_test.rs
+++ b/tests/expression_parser_test.rs
@@ -33,18 +33,6 @@ fn test_if_expr_with_sep() {
 }
 
 #[test]
-fn test_const_assign() {
-    let result = parse_expr("X = 1");
-    assert_eq!(
-        result.unwrap(),
-        ast::assignment(
-            ast::const_ref(vec!["X".to_string()]),
-            ast::decimal_literal(1)
-        )
-    )
-}
-
-#[test]
 fn test_equality_expr() {
     let result = parse_expr("1 != 2");
     assert_eq!(

--- a/tests/sk/enum.sk
+++ b/tests/sk/enum.sk
@@ -1,11 +1,11 @@
-a = Maybe::Some<Int>.new(1)
-b = Maybe::None
+a = Some<Int>.new(1)
+b = None
 f = fn(x: Maybe<Int>) { x }
 f(a)
 f(b)
 unless a.value == 1; puts "ng Some#value"; end
 
-o = Result::Ok<Int, Error>.new(0)
-e = Result::Err<Int, Error>.new(Error.new("fail"))
+o = Ok<Int, Error>.new(0)
+e = Fail<Int, Error>.new(Error.new("fail"))
 
 puts "ok"

